### PR TITLE
Update how-expo-works.md

### PR DESCRIPTION
**Pull request by pensierinmusica**
Add details in documentation for production flags

Closes https://github.com/expo/expo/issues/507

### CODE REVIEW

File: versions/v20.0.0/guides/how-expo-works.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ The second purpose is to serve assets. When you include an image in your app, yo
 
 ## Publishing/Deploying an Expo app in Production
 
-When you Publish an Expo app, we compile it into a JavaScript bundle with production flags enabled (minify, disable runtime development checks) and upload that bundle, along with any assets that it requires (see [Assets](assets.html#all-about-assets)) to CloudFront. We also upload your [Manifest](#expo-manifest) (including most of your `app.json` configuration) to our server.
+When you Publish an Expo app, we compile it into a JavaScript bundle with production flags enabled (i.e. minified and with [`__DEV__`](https://facebook.github.io/react-native/docs/javascript-environment.html#polyfills) set to `false`) and upload that bundle, along with any assets that it requires (see [Assets](assets.html#all-about-assets)) to CloudFront. We also upload your [Manifest](#expo-manifest) (including most of your `app.json` configuration) to our server.
```
Comment (abi) [inline versions/v20.0.0/guides/how-expo-works.md]: @ide Is this true?

Comment (abi) [reply versions/v20.0.0/guides/how-expo-works.md]: I looked around the `xdl` code and couldn't find anywhere where `__DEV__` is set to false. Guessing the default is false?

Comment (ide) [reply versions/v20.0.0/guides/how-expo-works.md]: Setting `dev=false` in the packager tells it to internally set `__DEV__` to false. It also builds the bundle with NODE_ENV=production which could invoke a different set of Babel plugins so I don't think we should say only that `__DEV__` is set to false.

File: versions/v20.0.0/guides/how-expo-works.md [hunk 2]
```diff
@@ -79,7 +79,7 @@ The second purpose is to serve assets. When you include an image in your app, yo
 
 ## Publishing/Deploying an Expo app in Production
 
-When you Publish an Expo app, we compile it into a JavaScript bundle with production flags enabled (minify, disable runtime development checks) and upload that bundle, along with any assets that it requires (see [Assets](assets.html#all-about-assets)) to CloudFront. We also upload your [Manifest](#expo-manifest) (including most of your `app.json` configuration) to our server.
+When you Publish an Expo app, we compile it into a JavaScript bundle with production flags enabled (i.e. we minify the source, and we tell the React Native packager to build in production mode, which in turn sets [`__DEV__`](https://facebook.github.io/react-native/docs/javascript-environment.html#polyfills) to `false` amongst other things) and upload that bundle, along with any assets that it requires (see [Assets](assets.html#all-about-assets)) to CloudFront. We also upload your [Manifest](#expo-manifest) (including most of your `app.json` configuration) to our server.
```
Comment (abi) [inline versions/v20.0.0/guides/how-expo-works.md]: Does this sound better? @ide

Comment (ide) [reply versions/v20.0.0/guides/how-expo-works.md]: Reads mostly ok to me, I might suggest removing the first comma in the parenthetical. The content seems fine though.

Comment (abi) [reply versions/v20.0.0/guides/how-expo-works.md]: Sweet, thanks!

**Comment by abi** [comment]
Merged into the main repo. Will show up on the docs site shortly. Thanks for your contribution, @pensierinmusica!

**Comment by pensierinmusica** [comment]
Awesome job everyone, thanks for the clarifications!  :)

@abi @ide

**Pull request closed without merging**